src/radiation/radiation_system.hpp

```diff
@@ -88,6 +89,29 @@ struct RadPressureResult {
 	double S;		       // maximum wavespeed for the radiation system
 };
 
+template <typename problem_t> struct NewtonIterationResult {
+	double Egas;									      // gas internal energy
+	double T_gas;									      // gas temperature
+	double T_d;									      // dust temperature
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> EradVec;		      // radiation energy density
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaPVec;		      // Planck mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaEVec;		      // energy mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaFVec;		      // flux mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> work;		      // work term
+	amrex::GpuArray<double, Physics_Traits<problem_t>::nGroups> delta_nu_kappa_B_at_edge; // Delta (nu * kappa_B * B)
+};
+
+// A struct to hold the results of ComputeJacobianForPureGas or ComputeJacobianForGasAndDust
+template <typename problem_t> struct JacobianResult {
+	double J00;
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> J0g;
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> Jg0;
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> Jgg;
+	double F0;
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> Fg;
+	double Fg_abs_sum;
```

Comment:
It's usually best to group the struct members of the same type together for optimal memory packing, since C++ requires that struct members are laid out in memory in the order they appear in the definition. So I would put all of the `double`s first.

Reply:
Done.